cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1032,6 +1033,33 @@ void reader::impl::decode_page_headers(hostdevice_vector<gpu::ColumnChunkDesc>&
   pages.device_to_host(stream, true);
 }
 
+__global__ void decompress_check_kernel(device_span<gpu_inflate_status_s> stats,
+                                        bool* any_block_failure)
+{
+  auto tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < stats.size()) {
+    if (stats[tid].status != 0) {
+      any_block_failure[0] = true;  // Doesn't need to be atomic
+    }
+  }
+}
+
+void decompress_check(device_span<gpu_inflate_status_s> stats,
+                      rmm::cuda_stream_view stream,
+                      bool* any_block_failure)
+{
+  dim3 block(128);
+  dim3 grid(cudf::util::div_rounding_up_safe(stats.size(), static_cast<size_t>(block.x)));
+  decompress_check_kernel<<<grid, block, 0, stream.value()>>>(stats, any_block_failure);
+}
+
+__global__ void convert_nvcomp_status(nvcompStatus_t* nvcomp_stats,
+                                      device_span<gpu_inflate_status_s> stats)
+{
+  auto tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < stats.size()) { stats[tid].status = static_cast<uint32_t>(nvcomp_stats[tid]); }
```

Comment:
While I understand this works because `nvcompSuccess == 0`, it would be better if this was checking specifically for `nvcompSuccess`. Something like `stats[tid].status = nvcomp_stats[tid] == nvcompSuccess? 0 : 1;`

Reply:
Yeah I agree this is safer - I have updated with this fix.